code/processes/dqc.q

```diff
@@ -29,9 +30,23 @@ init:{
   update starttime:.z.d+starttime from `.dqe.configtable;                                                       /- from timespan to timestamp
   update endtime:?[0W=endtime;0Wp;.z.d+endtime] from `.dqe.configtable;
 
-  .dqe.loadtimer'[.dqe.configtable]
+  .dqe.loadtimer'[.dqe.configtable];
+
+  .dqe.tosavedown:();                                                                                           /- store i numbers of rows to be saved down to DB
+  st:.dqe.writedownperiod+exec min starttime from .dqe.configtable;
+  et:$[0<'a:exec max endtime from (select from .dqe.configtable where checkid<>1) where (endtime<>0Wp);         /- checks that there is and end time in .dqe.configtable
```

Comment:
I think this might work out the same:
```
  et:(.eodtime.nextroll;a)0<'a:exec max endtime from .dqe.configtablewhere checkid<>1,endtime<>0Wp;
  et-:dqe.writedownperiod
```